lib/carrierwave/processing/vips.rb

```diff
@@ -0,0 +1,330 @@
+    def resize_to_fill(width, height, _gravity = nil, combine_options: {}, &block)
+      width, height = resolve_dimensions(width, height)
+
+      vips!(block) do |builder|
+        builder.resize_to_fill(width, height).apply(combine_options)
+      end
+    end
+
+    ##
+    # Resize the image to fit within the specified dimensions while retaining
+    # the original aspect ratio. If necessary, will pad the remaining area
+    # with the given color, which defaults to transparent (for gif and png,
+    # white for jpeg).
+    #
+    # See https://libvips.github.io/libvips/API/current/libvips-conversion.html#VipsCompassDirection
+    # for gravity options.
+    #
+    # === Parameters
+    #
+    # [width (Integer)] the width to scale the image to
+    # [height (Integer)] the height to scale the image to
+    # [background (List, nil)] the color of the background as a RGB, like [0, 255, 255], nil indicates transparent
+    # [gravity (String)] how to position the image
+    # [alpha (Boolean, nil)] pad the image with the alpha channel if supported
+    # [combine_options (Hash)] additional vips options to apply before resizing
+    #
+    # === Yields
+    #
+    # [Vips::Image] additional manipulations to perform
+    #
+    def resize_and_pad(width, height, background=nil, gravity='centre', alpha=nil, combine_options: {}, &block)
+      width, height = resolve_dimensions(width, height)
+
+      vips!(block) do |builder|
+        builder.resize_and_pad(width, height, background: background, gravity: gravity, alpha: alpha)
+          .apply(combine_options)
+      end
+    end
+
+    ##
+    # Returns the width of the image in pixels.
+    #
+    # === Returns
+    #
+    # [Integer] the image's width in pixels
+    #
+    def width
+      vips_image.width
+    end
+
+    ##
+    # Returns the height of the image in pixels.
+    #
+    # === Returns
+    #
+    # [Integer] the image's height in pixels
+    #
+    def height
+      vips_image.height
+    end
+
+    ##
+    # Manipulate the image with vips. This method will load up an image
+    # and then pass each of its frames to the supplied block. It will then
+    # save the image to disk.
+    #
+    # NOTE: This method exists mostly for backwards compatibility, you should
+    # probably use #vips!.
+    #
+    # === Gotcha
+    #
+    # This method assumes that the object responds to +current_path+.
+    # Any class that this module is mixed into must have a +current_path+ method.
+    # CarrierWave::Uploader does, so you won't need to worry about this in
+    # most cases.
+    #
+    # === Yields
+    #
+    # [Vips::Image] manipulations to perform
+    #
+    # === Raises
+    #
+    # [CarrierWave::ProcessingError] if manipulation failed.
+    #
+    def manipulate!
+      cache_stored_file! if !cached?
+      image = ::Vips::Image.open(current_path)
+
+      image = yield(image)
+      FileUtils.mv image.path, current_path
+
+      image.run_command("identify", current_path)
```

Comment:
Does this really work?
Seems like to be a leftover from CarrierWave::Processing::MiniMagick.

Reply:
IIRC I did left it there because my initial goal was to replicate the same interface as the `MiniMagick` module. I added a note in the module's documentation:

>     # NOTE: This method exists mostly for backwards compatibility, you should
>     # probably use #vips!.